html/dom/reflection.js

```diff
@@ -722,31 +722,45 @@ ReflectionTests.doReflects = function(data, idlName, idlObj, domName, domObj) {
     }
 
     for (var i = 0; i < idlTests.length; i++) {
+        var valueDesc = ReflectionHarness.stringRep(idlTests[i]);
         if ((data.type == "limited long" && idlTests[i] < 0) ||
             (data.type == "limited unsigned long" && idlTests[i] == 0)) {
             ReflectionHarness.testException("INDEX_SIZE_ERR", function() {
                 idlObj[idlName] = idlTests[i];
-            }, "IDL set to " + ReflectionHarness.stringRep(idlTests[i]) + " must throw INDEX_SIZE_ERR");
+            }, "IDL set to " + valueDesc + " must throw INDEX_SIZE_ERR");
         } else {
-            ReflectionHarness.run(function() {
```

Comment:
I don't think the approach of this patch is correct.  I think the problem is that when we call ReflectionHarness.test() inside ReflectionHarness.run(), it should call only assert_equals(), not test() (because ReflectionHarness.run() already calls test()).  Either modify ReflectionHarness.test() in new-harness.js to check if we're already inside a run() and if so don't call test(), or introduce a new ReflectionHarness.assertEquals() that's like ReflectionHarness.test() but without calling test() in new-harness.js, or something like that.  The way you do it here breaks reflection-original.html AFAICT -- there should be no testharness.js-specific code in reflection.js.

In the bigger picture, the right way to fix your problem is to condense test failures in a manner similar to how reflection-original.html does.  So perhaps instead of having a separate test() for every boolean attribute that if the content attribute is set to "foo" then the IDL property returns true, have a single test() that tests that for every attribute that's supposed to be boolean.  Or something like that.  I have some ideas on how to do this, but haven't figured them out in detail.  If this is blocking Blink/WebKit adopting this, maybe I could spend a couple of hours on it.